x86_energy_plugin.cpp

```diff
@@ -0,0 +1,682 @@
+        std::transform(sensor_name.begin(), sensor_name.end(), 
+                sensor_name.begin(), ::tolower);
+        /* only for real sensors exist the map entry
+         * but also key values that didn't exist are accessible
+         * use lower case keys because of the performance in the measure
+         * thread*/
+        auto sensor_data = readings[sensor_name + handle.quantity()];
+
+        auto converter =
+            scorep_time_converter<local_clock::time_point>(local_start, local_stop, scorep_start, scorep_stop);
+        local_clock::duration duration_actual;
+        scorep::chrono::ticks scorep_start_actual;
+
+        duration_actual = local_stop - local_start;
+        scorep_start_actual = scorep_start;
+
+
+        // We must use double here to avoid too much precision loss e.g. from integers in microseconds
+        const double sampling_period = static_cast<double>(duration_actual.count()) / readings_size;
+
+        logging::trace() << "Using effective sampling period of " << sampling_period << " sysclock ticks (usually us)";
+
+        const auto index_to_scorep_ticks = [sampling_period, scorep_start_actual, converter](size_t index)
+        {
+            // TODO Add offset, needs further time computation
+            return scorep_start_actual +
+                   converter.to_ticks(local_clock::duration(static_cast<int64_t>(sampling_period * index)));
+        };
+
+        logging::trace() << "Reading " << readings_size << " values from sensor " << handle.name();
+        cursor.resize(readings_size);
+
+        double value = 0;
+        const double sampling_period_secs =  std::chrono::duration_cast
+            <std::chrono::duration<double>>(duration_actual).count() / readings_size;
+        /* start calculation and filling of the cursor */
+        for (int i = 0; i < readings_size; i++)
+        {
+            if (handle.name() != "BLADE")
+            {
+                /* scale the results with the first measured value */
+                value = sensor_data[i] - sensor_data[0];
+            }
+            else
+            {
+                if (handle.quantity() == "E")
+                {
+                    /* for energy the offset power has to be multiplied with 
+                     * the time 
+                     * the time is casted to seconds*/
+                    value = offset/1000 * sampling_period_secs * i;
+                }
+                else if (handle.quantity() == "P")
+                {
+                    /* and for power the offset has to be simply added */
+                    value = offset/1000;
+                }
+
+                for (auto point : readings)
+                {
+                    /* only add if the sensor name didn't contain core
+                     * or one number is negative (for blade) 
+                     * and if the quantity of the blade matches*/
+                    if (point.first.find("core") == std::string::npos and 
```

Comment:
Wow. I didn't know that!
And nobody seems to use them, very strange 😕 
Therefore, I still prefer `&&`.